File: EJemplovBasic/frmAcceso.vb
Public Class frmAcceso

    Function validar_Password()
        If (txtNombreUser.Text = "Supervisor" And txtPassword.Text = "456") Then
            MsgBox("Bienvenido",, "Medikalsistem23")
            Me.Hide()
            frmMenuPrincipal.Show()
            txtNombreUser.BackColor = Color.White
            txtPassword.BackColor = Color.White
        Else
            MsgBox("Usuario y contraseña incorrectos", "Medykalsystem23")
            txtNombreUser.Text = ""
            txtPassword.Text = ""
            txtNombreUser.BackColor = Color.Violet
            txtPassword.BackColor = Color.Violet

        End If
    End Function
    Private Sub btnAceptar_Click(sender As Object, e As EventArgs) Handles btnAceptar.Click
        validar_Password()
    End Sub

    Private Sub btnCancelar_Click(sender As Object, e As EventArgs) Handles btnCancelar.Click
        End
    End Sub


    Private Sub txtPassword_KeyDown(sender As Object, e As KeyEventArgs) Handles txtPassword.KeyDown
        If e.KeyCode = Keys.Enter Then
            validar_Password()
        End If
    End Sub
End Class

File: Ejemplos12CIParcial/frmAcceso.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Ejemplos12CIParcial
{
    public partial class frmAcceso : Form
    {
        public frmAcceso()
        {
            InitializeComponent();
        }

        private void btnAceptar_Click(object sender, EventArgs e)
        {
            if (txtNombreUser.Text == "Supervisor" & txtPassword.Text == "456")
            {
                MessageBox.Show("Bienvenido", "Medikalsystem23");
                //comentarios aqui explique
                // usamos codigo de C# instrucciones If
                this.Hide();
                frmMenuPrincipal frm = new frmMenuPrincipal();
                frm.Show();
                txtNombreUser.BackColor = Color.White;
                txtPassword.BackColor = Color.White;
            }
            else
            {
                MessageBox.Show("Usuario y contraseña incorrectos", "Medykalsystem23");
                txtNombreUser.Text = "";
                txtPassword.Text = "";
                txtNombreUser.Focus();
                txtNombreUser.BackColor = Color.Violet;
                txtPassword.BackColor = Color.Violet;

            }
        }
     

    }
}
